src/Bash.cpp

```diff
@@ -0,0 +1,297 @@
+//
+// Created by gal on 7/1/23.
+//
+
+#include <algorithm>
+#include <cstdlib>
+#include <iostream>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <cstring>
+
+#include "Bash.h"
+
+/**
+ * Close the read and write ends of a pipe.
+ * @param pipe The file descriptor array representing the pipe.
+ */
+void closePipes(const fd& pipe) {
+    if (pipe[0] != -1) {
+        close(pipe[0]);
+        close(pipe[1]);
+    }
+}
+
+/**
+ * Find the index of a redirection symbol (">" or "<") in the given argument list.
+ * @param args The argument list.
+ * @param direction The redirection symbol ("<" or ">").
+ * @return The index of the redirection symbol in the argument list, or 0 if not found.
+ */
+size_t findRedirection(const vector<string>& args, const string& direction) {
+    auto it = std::find(args.begin(), args.end(), direction);
+    if (it != args.end()) {
+        return std::distance(args.begin(), it);
+    }
+    return 0;
+}
+
+/**
+ * Split a string into a vector of substrings using the specified delimiter.
+ * @param str The input string.
+ * @param delim The delimiter string.
+ * @return A vector of substrings.
+ */
+vector<string> splitString(string str, const string& delim) {
+    vector<string> words;
+    size_t pos;
+    while ((pos = str.find(delim)) != std::string::npos) {
+        words.emplace_back(str.substr(0, pos));
+        str.erase(0, pos + delim.length());
+    }
+    words.emplace_back(str.substr(0, pos));
+    return words;
+}
+
+/**
+ * Constructor for the Bash class.
+ * Initializes the pathDirectories vector by splitting the PATH environment variable.
+ * Sorts the pathDirectories vector in ascending order.
+ */
+Bash::Bash() {
+    for (auto &path : splitString(PATH, ":")) {
+        _pathDirectories.emplace_back(path);
+    }
+    std::sort(_pathDirectories.begin(), _pathDirectories.end(), [](path& s1, path& s2) {
```

Comment:
The original order of the paths in PATH is decided by the user and all standard programs follow that order. This is the way for the user to control which paths will be searched first and is useful in many cases. So I don't think it makes sense to sort the paths